cdap-api/src/main/java/co/cask/cdap/api/plugin/PluginConfig.java

```diff
@@ -34,6 +35,10 @@
   private PluginProperties properties;
   private Set<String> macroFields;
 
+  public PluginConfig() {
+    this.macroFields = Collections.emptySet();
```

Comment:
Right, sorry that's what I meant... that makes sense then, since there are no checks for `pluginConfig.getProperties() == null`